python/lsst/ts/ATHexapod/controller.py

```diff
@@ -0,0 +1,502 @@
+import asyncio
+import logging
+
+
+class ATHexapodController:
+    def __init__(self, log=None):
+
+        self.host = '127.0.0.1'
```

Comment:
Please explain how you can hard-code the host as the local host (preferably in a code comment). I thought this class was intended to talk to the low-level controller, in which case I would expect the host to be a constructor argument. Or does this code run on the same machine as the low-level controller?

Reply:
Well the controller gets handed its configuration from the CSC before it attempts to connect.

Reply:
Good question. The answer is definitely yes as long as the following is true:
* You connect only when in Disabled or Enabled state (or only Enabled, if you prefer, but I recommend the former). The recommended way to do this is in the `handle_summary_state` in an `if self.disabled_or_enabled:` block which connects if true and disconnects if false.
* You do not construct the CSC with initial_summary_state=disabled or enabled. I should probably make ConfigurableCsc smarter, but at the moment you have to transition using the usual state transition commands to reliably get configuration.

Reply:
Right. It's handled by the end_start and end_standby hooks.
And in production, you'll always start a CSC in Standby State with careful consideration for unit tests.

Reply:
Good. Though I urge you to consider moving it to `handle_summary_state` because what matters for deciding whether or not to connect is what state you are going into, not how you got into that state. Some states can be entered via more than one command.